core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala

```diff
@@ -61,7 +61,8 @@ class TimeSeriesPartition(val partID: Int,
                           // Shared class for mutating the infoMap / OffheapLFSortedIDMap given mapPtr above
                           offheapInfoMap: OffheapLFSortedIDMapMutator,
                           // Lock state used by OffheapLFSortedIDMap.
-                          var lockState: Int = 0)
+                          var lockState: Int = 0,
+                          var lastTime: Long = -1L)
```

Comment:
Looks like there may be an issue when a series is paged in. `lastTime` would be -1 and an out-of-order sample may potentially be introduced. You need to add logic to update lastTime once a partition is paged in.

Reply:
There are a couple of potential solutions, but they are all expensive.  We can retrieve the last ChunkSetInfo instead of the entire chunk, but even this is likely to add say 10-15 min to recovery.   We can also add a separate table which organizes the last time more efficiently.

However, if we are going to move to a more rigorous out of order solution, this might be moot.  So, perhaps we should just add a note and TODO that we have this edge case, and merge this to see how big of a problem the out of order is in the first place.

Reply:
How about save the "lastTime" into CheckPoint? Should be easy to get that during ChunkSink time. It's at the group level rather than part level, but assumption is data comes in order anyways.

Reply:
The above would only work if most of the lastTimes in each TSPartition being flushed were about the same.  What if there is one with lagging timestamps?

Reply:
@velvia Timestamp to be check pointed should be the "lastTime" of all TSPartitions in that group for that time bucket. So "one with lagging timestamps" means that "lastTime of that TSParition" is <= "lastTime of the group time bucket", which is fine.
E.g. timeBucket cut time is 10:00, and lastTime of the timeBucket is also 10:00, and lastTime of TSPartition X is 9:30, which means there's no data for TSPartition X between 9:30 - 10:00. So when it starts ingestion(or recover) for newer data, we can safely assume that new data of TSPartition X should be >= 10:00.